syntax = "proto3";

package statistics;

service StatisticsService {
    rpc SendStatistics (StatisticsCollection) returns (Response);

    rpc SignIn (SignInCredentials) returns (Response);

    rpc SignUp (SignUpCredentials) returns (Response);
}

message StatisticsCollection {
    string player_id = 1;
    int32 round_id = 2;
    bool user_turn = 3;
    repeated string lost_territories = 4;
    map<string, int32> attacked_territories = 5;
}

message SignInCredentials {
    string player_id = 1;
    string password = 2;
}

message SignUpCredentials {
    string player_id = 1;
    string password = 2;
}


message Response {
    string message = 1;
    bool status = 2;
}
